### ISSUE TITLE
Minutiae Feature Extraction: AI Algorithm  

### ISSUE THREAD

**Issue by sauravkr-diat-ai** [archethic-foundation/biometrics-seed-extraction#14]
### Problem to solve
The previous approach has a common limitation i.e many minutiae points were not able to be detected . The goal here is to extract more accurate feature & look for robustness in feature extraction.
### Solution
Deep learning method is state of art approach feature extraction task. Our attempt is to look for appropriate DL architecture, develop & evaluate.

###
- [x] To test whether MinutiaeNet can be run by any feasible method & test the  performance [1][2].
**Output** - Performance analysis documentation.
- [x]  To develop Deep Learning architecture for minutiae extraction.
**->** Parallelly considering two best CNN based architecture to match project 
requirement, viz, F-Net [3] & customized model [4].  
**Output** - Documentation
- [x]  To Design architecture using Tensorflow & Keras.
**Output** -  Code + Documentation
- [x] [WIP] **Data preparation**
Currently, we are using publicly available FVC 2000, 2002, 2004 fingerprint data for training purpose. These dataset needs two fundamental operation before feeding into DL model i.e
i. [WIP] **Data Annotation** 
It is categorisation & labelling in the image for application.
[https://drive.google.com/file/d/1P5961_ooKqH120mOFWE6aaQvvj9letA6/view?usp=sharing](url)
Annotation of fingerprint data is critical. A sample example is shown above.
ii.[WIP] **Data augmentation** 
It increase the amount of data which helps the model to be robust. The goal here is to look for such an image augmentation technique which helps in data creation without loss of information.
The developed code is successfully able to generate 36 augmented image for each image.  It is under process to extend up to 54 images for each.

- [x] To write training code for the designed architecture.
- [x] Parameter Tuning :-
Hyperparameter value selected based on paper study i.e, LR - 2e-4, Dropout - 50, Batch size - 4, epoch-30

Refrences:
[1]. https://arxiv.org/abs/1712.09401
[2]. https://github.com/luannd/MinutiaeNet
[3]. https://ietresearch.onlinelibrary.wiley.com/doi/epdf/10.1049/iet-bmt.2019.0017
[4]. https://ieeexplore.ieee.org/document/9304861/

Conclusion:

**Comment by sauravkr-diat-ai** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Product-and-Design-Workspace-61efe71b9083a90018250d89/issues/archethic-foundation/biometrics-seed-extraction/14?planning-poker) with ZenHub @lucy-sha512 @nilesh-uniris @I-Archer-Zero

**Comment by AmodD** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Product-and-Design-Workspace-61efe71b9083a90018250d89/issues/archethic-foundation/biometrics-seed-extraction/14?planning-poker) with ZenHub @sauravkr-diat-ai
